core/Tracker.php

```diff
@@ -94,6 +95,10 @@ public function main(Handler $handler, RequestSet $requestSet)
         try {
             $this->init();
             $handler->init($this, $requestSet);
+
+            $corsHandler = new CORSHandler();
```

Comment:
maybe we put this into `init`? Or before `init` out of the try/catch as it seems to not throw an exception?

Actually...  After a while of thinking it should be done right after `Tracker::loadTrackerEnvironment();` in `piwik.php` I reckon. Since `main()` might be also used when not being in a web context, and `piwik.php` makes it actually available to the web so it should be there. This keeps the code cleaner and better re-usable etc. The tracker class and `main` method is built to be used with different request and this keeps the responsibility to tracking and not to output handling (which is mainly done in `piwik.php`).

Reply:
Absolutely right :+1: made the change